mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -624,7 +640,16 @@ impl SaplingSyncLoopHandle {
                 },
                 None => self.notify_building_wallet_db(0, current_block.into()),
             }
-            scan_cached_blocks(&self.consensus_params, &self.blocks_db, &mut wallet_ops, Some(1000))?;
+
+            scan_cached_blocks(
+                &self.consensus_params,
+                &self.blocks_db,
+                &mut wallet_ops,
+                Some(self.scan_blocks_per_iteration),
+            )?;
+            if self.scan_interval_ms > 0 {
+                std::thread::sleep(Duration::from_millis(self.scan_interval_ms));
```

Comment:
If I understand correctly, `SaplingSyncLoopHandle::scan_blocks` is called within a future spawned asynchronously.
Can this be replaced with `Timer::sleep`?

Reply:
`SaplingSyncLoopHandle::scan_blocks` is not async. It does a lot of blocking DB interactions, so refactoring doesn't seem to make sense. It's used only wrapped in `block_in_place`:
https://github.com/KomodoPlatform/atomicDEX-API/blob/dd5911a360ec4083136288ea1d03fc6259afa737/mm2src/coins/z_coin/z_rpc.rs#L701
so `thread::sleep` looks acceptable here.

Reply:
I see now, thanks!